### PR TITLE
OTA MISRA warnings fix

### PR DESCRIPTION
**Pull request by tgsong**
Description
-----------

Fix MISRA warnings in OTA agent.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 1]
```diff
@@ -1075,7 +1109,12 @@ static OTA_Err_t prvProcessDataHandler( OTA_EventData_t * pxEventData )
         if( xResult == eIngest_Result_FileComplete )
         {
             /* File receive is complete and authenticated. Update the job status with the self_test ready identifier. */
-            xOTA_ControlInterface.prvUpdateJobStatus( &xOTA_Agent, eJobStatus_InProgress, ( int32_t ) eJobReason_SigCheckPassed, ( int32_t ) NULL );
+            xErr = xOTA_ControlInterface.prvUpdateJobStatus( &xOTA_Agent, eJobStatus_InProgress, ( int32_t ) eJobReason_SigCheckPassed, 0 );
```
Comment (ravibhagavandas) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1112]: Is type cast to `int32_t` required here ?

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 2]
```diff
@@ -1118,10 +1162,15 @@ static OTA_Err_t prvProcessDataHandler( OTA_EventData_t * pxEventData )
             /* We're actively receiving a file so update the job status as needed. */
             /* First reset the momentum counter since we received a good block. */
             xOTA_Agent.ulRequestMomentum = 0;
-            xOTA_ControlInterface.prvUpdateJobStatus( &xOTA_Agent, eJobStatus_InProgress, ( int32_t ) eJobReason_Receiving, ( int32_t ) NULL );
+            xErr = xOTA_ControlInterface.prvUpdateJobStatus( &xOTA_Agent, eJobStatus_InProgress, ( int32_t ) eJobReason_Receiving, 0 );
```
Comment (ravibhagavandas) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1165]: Is typecast to `int32_t` required here ?

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 3]
```diff
@@ -861,13 +864,19 @@ static OTA_Err_t prvRequestJobHandler( OTA_EventData_t * pxEventData )
...
             /* Send shutdown event to the OTA Agent task. */
             xEventMsg.xEventId = eOTA_AgentEvent_Shutdown;
-            OTA_SignalEvent( &xEventMsg );
 
-            /*
-             * Too many requests have been sent without a response or too many failures
-             * when trying to publish the request message. Abort. Store attempt count in low bits.
-             */
-            xReturn = ( uint32_t ) kOTA_Err_MomentumAbort | ( otaconfigMAX_NUM_REQUEST_MOMENTUM & ( uint32_t ) kOTA_PAL_ErrMask );
+            if( !OTA_SignalEvent( &xEventMsg ) )
```
Comment (gkwicker) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:868]: We're changing logic here in addition to fixing MISRA warnings, you may want to make sure that your commit message mentions this.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 4]
```diff
@@ -2410,7 +2487,7 @@ static IngestResult_t prvIngestDataBlock( OTA_FileContext_t * C,
     }
 
     /* Close the file and cleanup.*/
-    if( eIngestResult == eIngest_Result_Uninitialized )
+    if( eIngestResult == eIngest_Result_Accepted_Continue )
```
Comment (gkwicker) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2490]: Is `eIngest_Result_Accepted_Continue == eIngest_Result_Uninitialized`?  Seems like we've changed the logic here based on the new name.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 1 above)
Comment (tgsong) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1112]: yeah, seems redundant, will remove.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 3 above)
Comment (tgsong) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:868]: Actually not, I think it's GitHub confusing about the diff, I'm only catching the error returned from `OTA_SignalEvent`.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c (hunk 4 above)
Comment (tgsong) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2490]: this is an error made from Prasad previously, I'm correcting it here, see this commit, b5f218a5fbd905ff9624c44f958b49715064092b

**Pull request merged**